tracer/src/Datadog.Trace/IAST/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -0,0 +1,227 @@
+// <copyright file="HardcodedSecretsAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Analyzers;
+
+internal class HardcodedSecretsAnalyzer
+{
+    private const int UserStringsArraySize = 100;
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<HardcodedSecretsAnalyzer>();
+    private static HardcodedSecretsAnalyzer? _instance = null;
+
+    private static bool _started = false;
+
+    private static List<SecretRegex>? _secretRules = null;
+
+    private static ManualResetEventSlim _waitEvent = new ManualResetEventSlim(false);
+
+    public HardcodedSecretsAnalyzer()
+    {
+        Log.Debug("HardcodedSecretsAnalyzer -> Init");
+        LifetimeManager.Instance.AddShutdownTask(RunShutdown);
+        _started = true;
+        Task.Run(() => PoolingThread());
+    }
+
+    private static void PoolingThread()
+    {
+        try
+        {
+            Log.Debug("HardcodedSecretsAnalyzer polling thread -> Started");
+            while (_started)
+            {
+                var userStrings = new UserStringInterop[UserStringsArraySize];
+                int userStringLen = NativeMethods.GetUserStrings(userStrings.Length, userStrings);
+                Log.Debug("HardcodedSecretsAnalyzer polling thread -> Retrieved {UserStringLen} strings", userStringLen.ToString());
+                if (userStringLen > 0 && Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationId.HardcodedSecret))
```

Comment:
nit: I wonder if checking this here makes sense🤔 If the integration isn't enabled (initially) and subsequently _is_ enabled, then you would potentially have already "missed" all the strings, no? 

I would suggest moving this check _before_ you call `NativeMethods.GetUserStrings`? That way, even if it's initially disabled and _subsequently_ enabled, you still get the full analysis I think? 🤔

Reply:
Fixed in f76aadb89980cba30745182f464525418d9b5564